__test__/createPost.test.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { MantineProvider } from '@mantine/core';
+import { render, waitFor, screen, act } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import HomePage from '@/app/home/page';
+import { DataProvider } from '@/contexts/DataContext';
+import { Visibility } from '@/src/API';
+import userEvent from '@testing-library/user-event';
+import { cleanup } from '@testing-library/react';
+
+jest.mock('formik', () => ({
+    ...jest.requireActual('formik'),
+    useFormik: jest.fn().mockImplementation(() => ({
+      handleSubmit: jest.fn(),
+      handleChange: jest.fn(),
+      handleBlur: jest.fn(),
+      values: {
+        author: {
+            fname: 'John',
```

Comment:
Not sure if this affects how the code is being run but change this to firstName and lastName just in case